test/badsalt.c

```diff
@@ -433,6 +434,21 @@ test_one_case (const struct testcase *t,
       p = page + pagesize - (t->plen + 2);
       memcpy (p, goodhash, t->plen);
       p[t->plen] = '*';
+
+      const char *md5_label           = "MD5 (FreeBSD)";
+      const char *sha256_plain_label  = "SHA-256 (plain)";
+      const char *sha512_plain_label  = "SHA-512 (plain)";
+      const char *sha256_rounds_label = "SHA-256 (rounds)";
+      const char *sha512_rounds_label = "SHA-512 (rounds)";
+
+      /* The asterisk character is a valid salt character for these hashes.  */
+      if (!strncmp (t->label, md5_label, strlen (md5_label))                      ||
+          !strncmp (t->label, sha256_plain_label, strlen (sha256_plain_label))    ||
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/besser82/libxcrypt/pullRequest?prid=5767430)